common/lib/xmodule/xmodule/modulestore/split_mongo/split_draft.py

```diff
@@ -300,6 +301,69 @@ def get_orphans(self, course_key, **kwargs):
         course_key = self._map_revision_to_branch(course_key)
         return super(DraftVersioningModuleStore, self).get_orphans(course_key, **kwargs)
 
+    def delete_orphans(self, course_key, user_id, **kwargs):
+        draft_preferred_course_key = self._map_revision_to_branch(course_key)
+        published_course_key = self._map_revision_to_branch(
+            course_key, ModuleStoreEnum.RevisionOption.published_only
+        )
+
+        draft_orphans = super(DraftVersioningModuleStore, self).get_orphans(
+            draft_preferred_course_key, **kwargs
+        )
+
+        published_orphans = super(DraftVersioningModuleStore, self).get_orphans(
+            published_course_key, **kwargs
+        )
+
+        draft_orphan_ids = [orphan.block_id for orphan in draft_orphans]
+        published_orphan_ids = [orphan.block_id for orphan in published_orphans]
+
+        published_block_ids = [
+            block.id for block in self._lookup_course(published_course_key).structure['blocks']
+        ]
+
+        draft_block_ids = [
+            block.id for block in self._lookup_course(draft_preferred_course_key).structure['blocks']
+        ]
+
+        orphans = draft_orphans + published_orphans
+        # import ipdb; ipdb.set_trace()
+        if orphans:
```

Comment:
Oh, just saw the recursion at the bottom. The concern I have w/ the current setup is that it might blow up the stack, if there are a lot of orphans.